peer/peer_test.go

```diff
@@ -0,0 +1,384 @@
+// Copyright (c) 2015 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package peer_test
+
+import (
+	"io"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/btcsuite/btcd/addrmgr"
+	"github.com/btcsuite/btcd/blockchain"
+	"github.com/btcsuite/btcd/peer"
+	"github.com/btcsuite/btcd/wire"
+)
+
+// Mocks a network connection. Implements the net.Conn interface
+type Conn struct {
```

Comment:
I see you unexported `Pipe` but not this type and its fields.  Was that intentional?